mmv1/third_party/terraform/services/container/node_config.go.erb

```diff
@@ -896,11 +896,25 @@ func expandNodeConfig(v interface{}) *container.NodeConfig {
 	}
 
 	if v, ok := nodeConfig["secondary_boot_disks"]; ok && len(v.([]interface{})) > 0 {
-		conf := v.([]interface{})[0].(map[string]interface{})
-		nc.SecondaryBootDisks = append(nc.SecondaryBootDisks, &container.SecondaryBootDisk{
-			DiskImage: conf["disk_image"].(string),
-			Mode:      conf["mode"].(string),
-		})
+		conf, confOK := v.([]interface{})[0].(map[string]interface{})
+		if confOK {
+			modeValue, modeOK := conf["mode"]
+			diskImage := conf["disk_image"].(string)
+			if modeOK && modeValue.(string) == "CONTAINER_IMAGE_CACHE" {
```

Comment:
What would you think about checking that it's not empty or equal to `MODE_UNSPECIFIED` here? That would make the code forwards-compatible with other potential future enum values.

Reply:
After clarifying with @elfinhe about intended Terraform behavior, I removed mode changing logic. This check against CONTAINER_IMAGE_CACHE is no longer used. The new logic just checks for modeOK, meaning a mode exists, so type assertion to string does not panic.